libs/use-multi-media-devices/src/index.ts

```diff
@@ -0,0 +1,474 @@
+  const [cameraList, setCameraList] = useState<InputDeviceInfo[]>([]);
+  const [microphoneList, setMicrophoneList] = useState<InputDeviceInfo[]>([]);
+
+  const _handleError = (error: unknown) => {
+    if (error instanceof Error) {
+      handleError?.(error.message);
+    } else {
+      handleError?.(`${error}`);
+    }
+  };
+
+  const getMediaDevicesLists = async (): Promise<MediaDevicesLists> => {
+    const microphoneList: InputDeviceInfo[] = [];
+    const cameraList: InputDeviceInfo[] = [];
+    try {
+      const devices = await navigator.mediaDevices.enumerateDevices();
+      devices.forEach((device) => {
+        if (device.kind === 'audioinput' && isUniqueDevice(microphoneList, device)) microphoneList.push(device);
+        else if (device.kind === 'videoinput' && isUniqueDevice(cameraList, device)) cameraList.push(device);
+      });
+      setRootCameraList(cameraList);
+      setRootMicrophoneList(microphoneList);
+    } catch (error: unknown) {
+      _handleError(error);
+    }
+    return Promise.resolve({ cameraList, microphoneList });
+  };
+
+  const addStream: MediaDevices['addStream'] = async ({
+    type,
+    microphone,
+    camera,
+    audioConstraints = {},
+    videoConstraints = {},
+  }) => {
+    try {
+      if (type === StreamTypes.MEDIA) {
+        if (microphone && camera) {
+          const constraints = {
+            audio: {
+              deviceId: { exact: microphone.deviceId },
+              ...audioConstraints,
+            },
+            video: {
+              deviceId: { exact: camera.deviceId },
+              ...idealCameraConfig,
+              ...videoConstraints,
+            },
+          };
+          const stream = await navigator.mediaDevices.getUserMedia(constraints);
+          const audioTracks = stream.getAudioTracks()[0];
+          const videoTracks = stream.getVideoTracks()[0];
+          const cameraCapabilities = videoTracks.getCapabilities();
+          dispatch({
+            type: StreamsActionType.ADD_STREAM,
+            id: stream.id,
+            stream: {
+              type,
+              display: stream,
+              capabilities: {
+                microphone: audioTracks.getCapabilities(),
+                camera: videoTracks.getCapabilities(),
+              },
+              resolutions: allResolutions.filter((resolution) => {
+                return (
+                  resolution.width <= (cameraCapabilities?.width?.max ?? 0) &&
+                  resolution.height <= (cameraCapabilities?.height?.max ?? 0)
+                );
+              }),
+              settings: {
+                microphone: audioTracks.getSettings(),
+                camera: videoTracks.getSettings(),
+              },
+              device: {
+                microphone,
+                camera,
+              },
+              state: initialStreamState,
+            },
+          });
+        }
+      } else {
```

Comment:
```suggestion
      } else if (type === StreamTypes.MEDIA)
```
Or use `switch` and `case`